csvvalidator.py

```diff
@@ -1059,21 +1067,20 @@ def write_problems(problems, file, summarize=False, limit=0):
                 underline += '-'
             underline += '\n'
             w(underline)
-            for k in sorted(p.viewkeys() - {'code', 'message', 'context'}):
+            for k in sorted(set(p.keys()) - {'code', 'message', 'context'}):
                 w(':%s: %s\n' % (k, p[k]))
             if 'context' in p:
                 c = p['context']
-                for k in sorted(c.viewkeys()):
+                for k in sorted(list(c.keys())):
```

Comment:
I'm curious why one of these is a list and on is a set

Reply:
The one that is a set is being combined with another set. This one probably doesn't even need to be wrapped in a list, but it's good practice for python3 since `keys` is a generator in python3 and a list in python2.

Reply:
should the change to keys below also be wrapped in a list then?